moodle/Sniffs/Access/DeprecatedCapabilitySniff.php

```diff
@@ -0,0 +1,111 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Checks that each file contains necessary login checks.
+ *
+ * @package    local_codechecker
+ * @copyright  2022 Laurent David <laurent.david@moodle.com>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace MoodleCodeSniffer\moodle\Sniffs\Access;
+
+use PHP_CodeSniffer\Files\File;
+use PHP_CodeSniffer\Sniffs\Sniff;
+
+class DeprecatedCapabilitySniff implements Sniff {
+
+    // phpcs:disable moodle.NamingConventions.ValidVariableName.MemberNameUnderscore
+    /**
+     * If we try to check this capability, a warning will be shown.
+     *
+     * @var array
+     */
+    public $capabilitiesWarningList = [];
+
+    /**
+     * If we try to check this capability, an error will be shown.
+     *
+     * @var array
+     */
+    public $capabilitiesErrorList = [];
+
+    // phpcs:enable
+
+    /**
+     * Access function type
+     *
+     * @var string[]
+     */
+    public $accessfunctions = ['has_access', 'has_capability'];
```

Comment:
What about `has_any_capability`, `has_all_capabilities`, and the `require_*` versions?